controllers/courses.js

```diff
@@ -717,8 +723,14 @@ router.get('/:courseId/', async (req, res, next) => {
 					: course.name,
 				activeTab: req.query.activeTab,
 				lessons,
-				homeworks: homeworks.filter(task => !task.private),
-				myhomeworks: homeworks.filter(task => task.private),
+				homeworks: { homeworks },
```

Comment:
Do you only use this for the length of the array? Maybe you could return the length directly here. (then with adapted identifier)